src/bin/pgcopydb/ld_stream.c

```diff
@@ -221,6 +366,106 @@ stream_init_specs(StreamSpecs *specs,
 }
 
 
+/*
+ * Converts source filters to wal2json options.
+ *
+ * This function takes a SourceFilters struct and two PQExpBuffer objects. It converts the filters
+ * into comma-separated lists of tables and schemas to include or exclude, and appends these lists
+ * to the provided PQExpBuffer objects.
+ *
+ * Returns true if the conversion was successful, false otherwise.
+ */
+bool
+convertFiltersToWal2JsonOptions(SourceFilters *filters,
+								PQExpBuffer commaSeperatedAddTables,
+								PQExpBuffer commaSeperatedFilterTables)
+{
+	if (filters->type == SOURCE_FILTER_TYPE_NONE)
+	{
+		return true;
+	}
+
+	for (int i = 0; i < filters->includeOnlyTableList.count; i++)
+	{
+		char *filteredNspName = filters->includeOnlyTableList.array[i].nspname;
+		char *filteredRelName = filters->includeOnlyTableList.array[i].relname;
+
+		if (commaSeperatedAddTables->len != 0)
+		{
+			appendPQExpBufferChar(commaSeperatedAddTables, ',');
+		}
+
+		appendPQExpBuffer(commaSeperatedAddTables, "%s.%s", filteredNspName,
+						  filteredRelName);
+	}
+
+	for (int i = 0; i < filters->includeOnlySchemaList.count; i++)
+	{
+		char *filteredNspName = filters->includeOnlySchemaList.array[i].nspname;
+
+		if (commaSeperatedAddTables->len != 0)
+		{
+			appendPQExpBufferChar(commaSeperatedAddTables, ',');
+		}
+
+		appendPQExpBuffer(commaSeperatedAddTables, "%s.*", filteredNspName);
+	}
+
+	for (int i = 0; i < filters->excludeTableList.count; i++)
+	{
+		char *filteredNspName = filters->excludeTableList.array[i].nspname;
+		char *filteredRelName = filters->excludeTableList.array[i].relname;
+
+		if (commaSeperatedFilterTables->len != 0)
+		{
+			appendPQExpBufferChar(commaSeperatedFilterTables, ',');
+		}
+
+		appendPQExpBuffer(commaSeperatedFilterTables, "%s.%s", filteredNspName,
+						  filteredRelName);
+	}
+
+	for (int i = 0; i < filters->excludeSchemaList.count; i++)
+	{
+		char *filteredNspName = filters->excludeSchemaList.array[i].nspname;
+
+		if (commaSeperatedFilterTables->len != 0)
+		{
+			appendPQExpBufferChar(commaSeperatedFilterTables, ',');
+		}
+
+		appendPQExpBuffer(commaSeperatedFilterTables, "%s.*", filteredNspName);
+	}
+
+	for (int i = 0; i < filters->excludeTableDataList.count; i++)
+	{
+		char *filteredNspName = filters->excludeTableDataList.array[i].nspname;
+		char *filteredRelName = filters->excludeTableDataList.array[i].relname;
+
+		if (commaSeperatedFilterTables->len != 0)
+		{
+			appendPQExpBufferChar(commaSeperatedFilterTables, ',');
+		}
+
+		appendPQExpBuffer(commaSeperatedFilterTables, "%s.%s", filteredNspName,
+						  filteredRelName);
+	}
+
+	if (PQExpBufferBroken(commaSeperatedAddTables) ||
+		PQExpBufferBroken(commaSeperatedFilterTables))
+	{
+		log_error("Failed to convert source filters to wal2json filters: out of memory");
+
+		destroyPQExpBuffer(commaSeperatedAddTables);
+		destroyPQExpBuffer(commaSeperatedFilterTables);
```

Comment:
I would rather leave the caller to deallocate these because it was allocated by the caller.